### DISCUSSION TITLE
Issue with yt helper? - Transcript not available. (transcript not found)

### DISCUSSION THREAD

**Discussion by exponentiallyio** [danielmiessler/Fabric#925] [Q&A] [answered]
I've just installed fabric and the yt helper, but when I run:

yt --transcript https://www.youtube.com/watch?v=bhnfZhJWCWY

I get:
Transcript not available. (transcript not found)

it's the same with every video. I've inserted by Youtube API key

Thanks for any help!

**Comment by Krizzology** [comment]
I have had the same thing,,,,,,,Watching for a response on this channel. In addition I also get this error: Anybody that offer some help regarding the vendor?

(fabric) PS C:\Users\Christopher Jooste> yt https://youtu.be/w2gbOQOJ2BQ?si=s_fZ1d0ji5x5R_DB | fabric --pattern analyze_answers
could not find vendor.
 Model =
 DefaultModel = gpt-4
 DefaultVendor = openai
(fabric) PS C:\Users\Christopher Jooste> fabric /vendor: openai 
unknown flag `vendor'

**Comment by oort77** [comment]
Same here: "Transcript not available. (transcript not found)"

**Comment by drlakshan** [comment]
Hi 
use the youtube link inside quotes. it works
<img width="975" alt="image" src="https://github.com/user-attachments/assets/c8cf95b1-354b-439a-a058-5ea1b097c11a">

**Reply by saxster** [reply]
It's not working even with the quotes like you suggest.
Regards

**Comment by alexxmugg** [comment]
https://github.com/danielmiessler/yt

**Comment by saxster** [comment]
Asked Claude to look at the code, here are the comments:

Potential errors:

The code assumes the presence of a .env file in a specific location (~/.config/fabric/.env). If this file doesn't exist or is inaccessible, the program will crash.
The transcript extraction method might be unreliable, as it depends on parsing the HTML of the YouTube page. YouTube's front-end structure can change, potentially breaking this functionality.
Error handling for API requests could be more robust. Currently, if comment fetching fails, it logs the error but continues execution.
The program doesn't handle rate limiting or quota exhaustion from the YouTube API, which could lead to unexpected behavior if limits are exceeded.

Suggestions for improvement:

Error handling: Implement more graceful error handling, especially for API calls and file operations.
Configuration: Use a more flexible configuration system, perhaps allowing users to specify the location of the .env file or use command-line arguments for the API key.
Modularization: Break down the large mainFunction into smaller, more focused functions to improve readability and maintainability.
Concurrency: Implement concurrent fetching of transcript, comments, and video details to improve performance for videos with many comments.
Testing: Add unit tests for critical functions like getVideoID, parseDuration, etc.
Input validation: Add more robust input validation, especially for the YouTube URL.
Pagination: Implement pagination for comment fetching to handle videos with more than 100 comments.
Caching: Consider implementing a caching mechanism for API responses to reduce API usage and improve performance for repeated queries.
Transcript reliability: Explore more reliable methods for transcript extraction, possibly using YouTube's official caption API if available.
Use context: Implement context for HTTP requests to allow for timeouts and cancellation.
Structured logging: Replace log.Fatal calls with a more structured logging approach that allows for better error reporting and doesn't abruptly terminate the program.
Implement custom errors: Define custom error types for different error scenarios to provide more meaningful error messages.

Overall, the code is functional but could benefit from improvements in robustness, performance, and maintainability. The suggestions above would help make the code more resilient and easier to maintain and extend in the future.

**Comment by leerogers** [comment]
v1.4.196

For me, it seems intermittent.  I ran yt https://youtu.be/GoMeXbsRPwA?si=I341MXx5RITmhmlO 10 times, twice came back with transcript, the other eight as "transcript not available. (EOF)".  So either Google API is throttling or yt is timing out maybe?

**Reply by ksylvan** [reply]
1.4.196 is an old version. Update to the latest and it will work. Also make sure to have the latest yt-dlp installed.

**Comment by miccid** [comment, marked as answer]
Fabric uses yt-dlp, I did this on my Ubuntu and got it fixed:

The Complete Installation Guide

There are two essential steps: removing the old version and installing the new one.

Step 1: Uninstall the Outdated apt Version

First, completely remove the version of yt-dlp that was installed by the package manager. This prevents any conflicts.

sudo apt remove yt-dlp

Step 2: Install the Latest Official Version

Next, download the latest release directly from the yt-dlp developers and make it executable. This is the official recommended installation method.

Run these two commands in your terminal one after the other:

# Command to download the latest binary
sudo wget https://github.com/yt-dlp/yt-dlp/releases/latest/download/yt-dlp -O /usr/local/bin/yt-dlp

# Command to make the binary executable
sudo chmod a+rx /usr/local/bin/yt-dlp

Stored here:

/usr/local/bin/yt-dlp

📌How to Update in the Future

The best part about this method is that updating is now simple. Because you are no longer using the apt version, the built-in updater will work. To update yt-dlp to the latest version at any time, just run:

sudo yt-dlp -U

This will ensure you always have the most recent fixes for services like YouTube and fabric.

**Reply by ksylvan** [reply]
You can also follow the instructions here: <https://github.com/yt-dlp/yt-dlp/wiki/Installation> for your specific OS to get the latest packaged binary. For Debian-based systems, I recommend you add the yt-dlp repository mentioned in those instructions.

**Comment by ksylvan** [comment]
This is an old issue.

I recently removed the fragile webscraping code, which required us to duplicate the work already being done by the yt-dlp team, and replaced it with directly calling yt-dlp.

One caveat, noted in the README, is that your yt-dlp binaty must be new (at least 2025.06.09, preferrably 2025.06.25 or later).

**Reply by carias-rh** [reply]
Still the same, issue, I'm trying with the example from the Readme

```
$ fabric -y "https://youtube.com/watch?v=uXs-zPc63kM" --stream --pattern extract_wisdom                                                                              
transcript not available. (EOF)
$ yt-dlp --version 
2025.06.25
$ fabric --version                                                                                                                                     
v1.4.192

$ cat /etc/os-release 
NAME="Fedora Linux"
VERSION="42 (Workstation Edition)"
RELEASE_TYPE=stable
ID=fedora
VERSION_ID=42
VERSION_CODENAME=""
PLATFORM_ID="platform:f42"
PRETTY_NAME="Fedora Linux 42 (Workstation Edition)"
ANSI_COLOR="0;38;2;60;110;180"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:42"
DEFAULT_HOSTNAME="fedora"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f42/system-administrators-guide/"
SUPPORT_URL="https://ask.fedoraproject.org/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=42
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=42
SUPPORT_END=2026-05-13
VARIANT="Workstation Edition"
VARIANT_ID=workstation

```

**Reply by leerogers** [reply]
I tried the same commands it worked for me.  The only difference in setup was my fabric version.  Maybe something to try.

$ fabric --version
v1.4.202

**Reply by ksylvan** [reply]
Update your fabric. The version you are using is very old and uses old (no longer effective) web scraping code.

**Reply by ksylvan** [reply]
Even 1.4.202 is old. Many many improvements in the last few months.

**Reply by leerogers** [reply]
Done, thank you for all the work on this.

**Reply by ksylvan** [reply]
@carias-rh let me know it's working for you.

**Reply by carias-rh** [reply]
Working! That was easy ;)

**Comment by prakashgp** [comment]
I have been using this [youtube transcript scraper](https://apify.com/supreme_coder/youtube-transcript-scraper?fpr=ve081&fp_sid=github) on apify which is very cheap and quite easy to use. I've create a n8n workflow to convert youtube videos into twitter threads
